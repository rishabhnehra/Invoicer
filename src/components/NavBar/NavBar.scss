.NavBar {
    width: 100%;
    color: #FFF;
    background-color: #574B90;

    .container {
        display: flex;
        justify-content: space-between;
        max-width: 1000px;
        margin: 0 auto;

        .left-side {
            padding: 18px 32px;
        }

        .right-side {
            padding: 18px 32px;
            
            a:first-child {
                margin-right: 32px;
            }
        }
    }
}