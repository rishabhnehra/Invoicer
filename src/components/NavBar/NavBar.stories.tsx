import {Story, Meta} from '@storybook/react/types-6-0';

import { NavBar as NavBarComponent, NavBarProps } from './NavBar';

export default {
    title: 'Components/Navigation Bar',
    component: NavBarComponent
} as Meta;

const Template: Story<NavBarProps> = (args) => <NavBarComponent {...args} />

export const NavBar = Template.bind({});
NavBar.args = {
    title: 'My Dashboard',
}