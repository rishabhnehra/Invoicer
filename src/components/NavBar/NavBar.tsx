import './NavBar.scss';

export type NavBarProps = {
    title: string
}

export const NavBar: React.FC<NavBarProps> = ({title}) => <div className="NavBar">
    <div className="container">
        <div className="left-side">
            <h3>{title}</h3>
        </div>
        <div className="right-side">
            <a>My Account</a>
            <a>Logout</a>
        </div>
    </div>
</div>