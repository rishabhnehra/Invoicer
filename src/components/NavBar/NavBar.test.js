import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { NavBar } from './NavBar';

describe('<NavBar />', () => {
    it('snapshot check', () => {
        const tree = renderer.create(<NavBar title='My Dashboard' isLoggedIn />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})