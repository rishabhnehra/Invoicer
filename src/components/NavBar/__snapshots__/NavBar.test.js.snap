// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> snapshot check 1`] = `
<div
  className="NavBar"
>
  <div
    className="container"
  >
    <div
      className="left-side"
    >
      <h3>
        My Dashboard
      </h3>
    </div>
    <div
      className="right-side"
    >
      <a>
        My Account
      </a>
      <a>
        Logout
      </a>
    </div>
  </div>
</div>
`;
